src/rsb/patterns/Reader.cpp

```diff
@@ -30,55 +30,35 @@
 
 #include <rsc/threading/SynchronizedQueue.h>
 
-#include "Handler.h"
+#include "../Factory.h"
+#include "../Handler.h"
 
-#include "eventprocessing/EventReceivingStrategy.h"
-#include "eventprocessing/InRouteConfigurator.h"
-
-using namespace std;
-
-using namespace boost;
+namespace rsb {
+namespace patterns {
 
-using namespace rsb::eventprocessing;
-using namespace rsb::transport;
+Reader::Reader(const Scope&             scope,
+               const ParticipantConfig& config) :
+    Participant(scope, config),
+    listener(getFactory().createListener(scope, config)) {
 
-namespace rsb {
+    this->listener->addHandler(
+            HandlerPtr(new EventFunctionHandler(boost::bind(&Reader::handle,
+                                                            this, _1))),
+            true);
+}
 
-Reader::Reader(const vector<InConnectorPtr>& connectors,
-               const Scope&                  scope,
-               const ParticipantConfig&      config) :
-    Participant(scope, config) {
-    initialize(connectors, scope, config);
+Reader::~Reader() {
+    this->listener.reset();
 }
 
 std::string Reader::getKind() const {
     return "reader";
 }
 
 const std::set<std::string> Reader::getTransportURLs() const {
-    return this->configurator->getTransportURLs();
+    return std::set<std::string>();
```

Comment:
Shouldn't this pass through to the listener? Otherwise, we might have to check the Python implementation.